sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/RowConverter.scala

```diff
@@ -3,246 +3,220 @@
-        val idx = mdata.getLong("idx").toInt
-        json.get(colName).flatMap(v => Option(v)).map(toSQL(_, ArrayType(et, containsNull = true))).collect {
-          case elemsList: Seq[_] if elemsList.indices contains idx => elemsList(idx)
-        } orNull
-      case StructField(name, dataType, _, _) =>
-        json.get(name).flatMap(v => Option(v)).map(toSQL(_, dataType)).orNull
+
+    def fromObjectNodeToRow(schema: StructType, objectNode: ObjectNode): Row = {
+        val values: Seq[Any] = convertStructToSparkDataType(schema, objectNode)
+        new GenericRowWithSchema(values.toArray, schema)
     }
-    new GenericRowWithSchema(values.toArray, schema)
-  }
-
-  // TODO: moderakh this method requires a rewrite
-  // scalastyle:off
-  // this is borrowed from old OLTP spark connector
-  def toSQL(value: Any, dataType: DataType): Any = {
-    Option(value).map { value =>
-      (value, dataType) match {
-        case (list: List[AnyRef@unchecked], ArrayType(elementType, _)) =>
-          null
-        case (_, struct: StructType) =>
-//          if(JSONObject.NULL.equals(value)) return null // TODO
-
-          val jsonMap: Map[String, AnyRef] = value match {
-            case doc: ObjectNode => documentToMap(doc)
-            case hm: util.HashMap[_, _] => hm.asInstanceOf[util.HashMap[String, AnyRef]].asScala.toMap
-          }
-          recordAsRow(jsonMap, struct)
-        case (_, map: MapType) =>
-          (value match {
-            case document: ObjectNode => documentToMap(document)
-            case _ => value.asInstanceOf[java.util.HashMap[String, AnyRef]].asScala.toMap
-          }).map(element => (toSQL(element._1, map.keyType), toSQL(element._2, map.valueType)))
-        case (_, array: ArrayType) =>
-          // TODO fixme
-          null
-//          if(!JSONObject.NULL.equals(value))
-//            value.asInstanceOf[java.util.ArrayList[AnyRef]].asScala.map(element => toSQL(element, array.elementType)).toArray
-//          else
-//            null
-        case (_, binaryType: BinaryType) =>
-          value.asInstanceOf[java.util.ArrayList[Int]].asScala.map(x => x.toByte).toArray
-        case _ =>
-          //Assure value is mapped to schema constrained type.
-          enforceCorrectType(value, dataType)
-      }
-    }.orNull
-  }
-  // scalastyle:on
-  ////
-
-
-  def rowToObjectNode(row: Row): ObjectNode = {
-    val objectNode: ObjectNode = objectMapper.createObjectNode();
-    row.schema.fields.zipWithIndex.foreach({
-      case (field, i) => {
-        objectNode.set(field.name, convertToJson(row.get(i), field.dataType, isInternalRow = false))
+
+    def fromRowToObjectNode(row: Row): ObjectNode = {
+        val objectNode: ObjectNode = objectMapper.createObjectNode()
+        row.schema.fields.zipWithIndex.foreach({
+            case (field, i) =>
+                field.dataType match {
+                    case NullType  | _ if row.isNullAt(i) => objectNode.putNull(field.name)
```

Comment:
can be added later after this PR. please make sure we add test coverage for null case in both conversion directions

Reply:
Added a new test for that explicit scenario called:

`"null in spark row" should "translate to ObjectNode"`